src/System.Diagnostics.StackTrace.Symbols/src/System/Diagnostics/StackTrace/Symbols.cs

```diff
@@ -0,0 +1,222 @@
+
+                                int sequencePointCount = 0;
+                                foreach (SequencePoint sequence in sequencePoints) {
+                                    sequencePointCount++;
+                                }
+
+                                if (sequencePointCount > 0) {
+                                    int[] offsets = new int[sequencePointCount];
+                                    int[] lines = new int[sequencePointCount];
+                                    int[] columns = new int[sequencePointCount];
+                                    DocumentHandle[] documents = new DocumentHandle[sequencePointCount];
+
+                                    int i = 0;
+                                    foreach (SequencePoint sequence in sequencePoints) {
+                                        offsets[i] = sequence.Offset;
+                                        lines[i] = sequence.StartLine;
+                                        columns[i] = sequence.StartColumn;
+                                        documents[i] = sequence.Document;
+                                        i++;
+                                    }
+
+                                    // Search for the correct IL offset
+                                    int j;
+                                    for (j = 0; j < sequencePointCount; j++) {
+
+                                        // look for the entry matching the one we're looking for
+                                        if (offsets[j] >= ilOffset) {
+
+                                            // if this offset is > what we're looking for, ajdust the index
+                                            if (offsets[j] > ilOffset && j > 0) {
+                                                j--;
+                                            }
+                                            break;
+                                        }
+                                    }
+
+                                    // If we didn't find a match, default to the last sequence point
+                                    if (j == sequencePointCount) {
+                                        j--;
+                                    }
+
+                                    while (lines[j] == SequencePoint.HiddenLine && j > 0) {
+                                        j--;
+                                    }
+
+                                    if (lines[j] != SequencePoint.HiddenLine) {
+                                        sourceLine = lines[j];
+                                        sourceColumn = columns[j];
+                                    }
+                                    sourceFile = reader.GetString(reader.GetDocument(documents[j]).Name);
+                                }
+                            }
+                            catch {
+                            }
```

Comment:
Yes, we don’t need the catch. The portable pdb reader functions shouldn’t throw any exceptions that are not already fatal.  I was being paranoid because if any of this code does throw an exception it may cause a recursive cycle of death.

mikem

From: noahfalk [mailto:notifications@github.com]
Sent: Monday, May 16, 2016 10:35 PM
To: dotnet/corefx <corefx@noreply.github.com>
Cc: Mike McLaughlin <mikem@microsoft.com>; Assign <assign@noreply.github.com>
Subject: Re: [dotnet/corefx] Add symbol support helper assembly for mscorlib's StackTrace. (#8591)


In src/System.Diagnostics.StackTrace.Symbols/src/System/Diagnostics/StackTrace/Symbols.cs<https://github.com/dotnet/corefx/pull/8591#discussion_r63466503>:

> +                                        j--;

> +                                    }

> +

> +                                    while (lines[j] == SequencePoint.HiddenLine && j > 0) {

> +                                        j--;

> +                                    }

> +

> +                                    if (lines[j] != SequencePoint.HiddenLine) {

> +                                        sourceLine = lines[j];

> +                                        sourceColumn = columns[j];

> +                                    }

> +                                    sourceFile = reader.GetString(reader.GetDocument(documents[j]).Name);

> +                                }

> +                            }

> +                            catch {

> +                            }

Do we need this catch here? I could understand wanting a catch in mscorlib around the entire operation, but putting a blanket catch around just this block wasn't clear what purpose it served.

—
You are receiving this because you were assigned.
Reply to this email directly or view it on GitHub<https://github.com/dotnet/corefx/pull/8591/files/9802644d90aa7fe6aba6d621724a307212303d08#r63466503>